runtime/drivers/s3/api.go

```diff
@@ -27,7 +26,7 @@ func (c Connector) ListBuckets(ctx context.Context, env *connectors.Env) ([]stri
 	}
 
 	sharedConfigState := session.SharedConfigDisable
-	if env.AllowHostAccess {
+	if val, ok := c.config["allow_host_access"]; ok && val.(bool) {
```

Comment:
1. type cast panic issue 
2. should we pass `allow_host_access` as an explicit param in `Open` (maybe have `opts *OpenOptions` for future params)? Maybe not worth it, don't know

Reply:
2. We can parse config to a structure and that should make it more explicit.

Reply:
I'm good either way. Parsing to a config also seems good